config/config.go

```diff
@@ -936,3 +1008,15 @@ func (c *RemoteReadConfig) UnmarshalYAML(unmarshal func(interface{}) error) erro
 	// Thus we just do its validation here.
 	return c.HTTPClientConfig.Validate()
 }
+
+func filePath(filename string) string {
+	absPath, err := filepath.Abs(filename)
+	if err != nil {
+		return filename
+	}
+	return absPath
+}
+
+func fileErr(filename string, err error) error {
+	return fmt.Errorf("%s: %w", filePath(filename), err)
```

Comment:
Maybe nicer when there are spaces etc. in a filename?

```suggestion
	return fmt.Errorf("%q: %w", filePath(filename), err)
```